src/components/syncplay/groupSelectionMenu.js

```diff
@@ -0,0 +1,198 @@
+                }
+                loading.hide();
+                return;
+            }
+
+            var menuOptions = {
+                title: globalize.translate('HeaderSyncplaySelectGroup'),
+                items: menuItems,
+                positionTo: button,
+                resolveOnClick: true,
+                border: true
+            };
+
+            actionsheet.show(menuOptions).then(function (id) {
+                if (id == "new-group") {
+                    apiClient.sendSyncplayCommand(sessionId, "NewGroup");
+                } else {
+                    apiClient.sendSyncplayCommand(sessionId, "JoinGroup", {
+                        GroupId: id,
+                        PlayingItemId: playingItemId
+                    });
+                }
+            }, emptyCallback);
+
+            loading.hide();
+        });
+    }).catch(function (error) {
+        console.error(error);
+        loading.hide();
+        toast({
+            text: globalize.translate('MessageSyncplayNoGroupsAvailable')
+        });
+    });
+}
+
+/**
+ * Used when user has joined a group.
+ * @param {HTMLElement} button - Element where to place the menu.
+ * @param {Object} user - Current user.
+ * @param {Object} apiClient - ApiClient.
+ */
+function showLeaveGroupSelection (button, user, apiClient) {
+    const sessionId = getActivePlayerId();
+    if (!sessionId) {
+        syncplayManager.signalError();
+        toast({
+            // TODO: translate
+            text: "Syncplay error occured."
+        });
+        return;
+    }
+
+
+    const menuItems = [{
+        name: globalize.translate('LabelSyncplayLeaveGroup'),
+        icon: "meeting_room",
+        id: "leave-group",
+        selected: true,
+        secondaryText: globalize.translate('LabelSyncplayLeaveGroupDescription')
+    }];
+
+    var menuOptions = {
+        title: globalize.translate('HeaderSyncplayEnabled'),
+        items: menuItems,
+        positionTo: button,
+        resolveOnClick: true,
+        border: true
+    };
+
+    actionsheet.show(menuOptions).then(function (id) {
+        if (id == "leave-group") {
+            apiClient.sendSyncplayCommand(sessionId, "LeaveGroup");
+        }
+    }, emptyCallback);
+
+    loading.hide();
+}
+
+// Register to Syncplay events
+let syncplayEnabled = false;
+events.on(syncplayManager, 'SyncplayEnabled', function (e, enabled) {
+    syncplayEnabled = enabled;
+});
+
+/**
+ * Shows a menu to handle Syncplay groups.
+ * @param {HTMLElement} button - Element where to place the menu.
+ */
+export function show (button) {
+    loading.show();
+
+    // TODO: should feature be disabled if playback permission is missing?
+    playbackPermissionManager.check().then(() => {
+        console.debug("Playback is allowed.");
+    }).catch((error) => {
+        console.error("Playback not allowed!", error);
+        toast({
+            text: globalize.translate("MessageSyncplayPlaybackPermissionRequired")
+        });
+    });
```

Comment:
This `PlaybackPermissionManager` should be adopted also when starting a normal playback.

Currently what happens is that the user clicks on an item they wish to play, the client sends a request to the server which in turn sends the play command asynchronously.

The problem is that in some browsers (mainly Safari I guess) the async play command on the media is seen as an autoplay initiated by the code and not by the user itself. So the play command is rejected with `NotAllowedError`.

The solution is to grab the permission inside a click event generated by the user, in this case when he clicks the `Sync` icon. Ideally this is needed once.

Grabbing permission simply means starting playing something inside the click event. This then allows the code to control the playback later on.

Reply:
this looks like a stuff to be put in separate PR...

Reply:
> this looks like a stuff to be put in separate PR...

Definitely.

For now `PlaybackPermissionManager` is used only when trying to enable SyncPlay, otherwise all the commands sent by the server through the WebSocket are blocked.